### PR TITLE
Modify Rule S2076(common text): Education Framework (APPSEC-54)

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
Specification ticket: [APPSEC-54](https://sonarsource.atlassian.net/browse/APPSEC-54).

### CODE REVIEW

File: rules/S2076/impact.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+In extreme cases, the attackers can either:
+
+* Remain on the server and download files over an extended period of time.
+* Send ransomware to the server to completely cripple it.
+
+This threat is particularly insidious if the attacked organization does not
+maintain a Disaster Recovery Plan (DRP).
+
+==== Chain OS injections with other vulnerabilities.
```
Comment (pierre-loup-tristant-sonarsource) [inline rules/S2076/impact.adoc:25]:
```suggestion
==== Root privilege escalation and rebound.
```

File: rules/S2076/common/fix/introduction.adoc [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Allowing users to execute operating system commands generally creates more
+problems than it solves.
+
+Anything that can be done via operating system commands can also be done via a
+language's native SDK. +
```
Comment (pierre-loup-tristant-sonarsource) [inline rules/S2076/common/fix/introduction.adoc:5]:
```suggestion
Anything that can be done via operating system commands can usually be done via a
language's native SDK. +
```

**Review by pierre-loup-tristant-sonarsource — CHANGES_REQUESTED** [review]
As discussed please rework the Impacts section.
I also added some improvement suggestions.

File: rules/S2076/impact.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
+=== What is the potential impact?
+In the context of a web application vulnerable to OS injection: +
+After detecting the injection, attackers inject data into the vulnerable field
```
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2076/impact.adoc:3]: This is an assumption that the attack vector is actually a web application field. I found that the context where people would identify the most

[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2076/impact.adoc:3]: I'm going to use `parameter` instead, to make it aligned with the code samples

[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2076/impact.adoc:3]: No, I know. I will add a small disclaimer saying that we assume it's a field

File: rules/S2076/impact.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+to execute malicious commands in the affected operating system(s).
+
+The impact depends on the access control measures taken on the target system
+OS.
+
+The following real-world scenarios illustrate the various impacts of an
+attacker exploiting the vulnerability.
```
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2076/impact.adoc:10]: I created APPSEC-65 for this. And I will revert to the old format

File: rules/S2076/impact.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2076/impact.adoc:3]: what about `text field` ? I made the modif a few minutes ago

[the remark this answers is not in the document]
Comment (pierre-loup-tristant-sonarsource) [reply rules/S2076/impact.adoc:3]: Yes, works too.

File: rules/S2076/rationale.adoc [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Applications that allow the execution of operating system commands from
+user-controlled data should control the command to execute. +
+Otherwise, an attacker can inject arbitrary commands that compromise the
+underlying operating system.
```
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2076/rationale.adoc:4]: I'm not a fan of sticking to patterns actually. I think it adds unnecessary constraints to our creativity when writing.

Do you think we should keep patterns in our texts for our users? Does it change something?

File: rules/S2076/common/fix/pre-approved-list.adoc [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+validate the input before using it in an OS command.
+
+Create a list of authorized and secure commands that you want the application
+to be able to execute. If a user input does not match an entry in this list, it
+should be rejected because it is considered unsafe.
+
+Depending on the number of commands you want the application to support, the
+list can be either a regex string or any array type. If you use regexes, choose
+simple regexes to avoid ReDOS attacks. For example, you can accept only Latin
+characters by using `^[a-zA-Z]{1-10}$`.
```
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2076/common/fix/pre-approved-list.adoc:14]: done

File: rules/S2076/rationale.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S2076/rationale.adoc:4]: To clarify : I'm doing the changes

File: rules/S2076/impact.adoc [hunk 7]
```diff
@@ -1,42 +1,43 @@
...
+maintain a disaster recovery plan (DRP).
 
-Depending on the underlying servers of the web application, the impact depends
-on how focused the target company is on its Defense In Depth.
+==== Root privilege escalation and pivot
 
-Disruptions include critical privilege escalation, enterprise-wide denials of
-service, or enterprise-wide data leaks. +
+In this scenario, the attacker can do everything described in the previous
+section. The kicker is that the attacker also manages to elevate his privileges
```
Comment (pierre-loup-tristant-sonarsource) [inline rules/S2076/impact.adoc:31]: "The kicker" seems familiar. What about "game changer"?

Comment (pierre-loup-tristant-sonarsource) [inline rules/S2076/impact.adoc:31]: "The kicker" seems familiar. What about "game changer"?

File: rules/S2076/impact.adoc [hunk 8]
```diff
@@ -1,42 +1,43 @@
...
+as an administrator and attack other servers.
 
-OS injections are truly gaping windows into the inner workings of an
-organization's information system.
+Depending on the underlying servers of the web application,
```
Comment (pierre-loup-tristant-sonarsource) [inline rules/S2076/impact.adoc:34]: This pending sentence looks like it should be removed.
```suggestion
```

**Review by pierre-loup-tristant-sonarsource — APPROVED** [review]
LGTM. Two minor fixes to apply to the impact section.

File: rules/S2076/impact.adoc (hunk 7 above)
Comment (loris-s-sonarsource) [reply rules/S2076/impact.adoc:31]: I didn't have the right word, but maybe even more simple would be "the difference", no ? "Game changer" looks like a happy impact

Comment (loris-s-sonarsource) [reply rules/S2076/impact.adoc:31]:
```suggestion
section. The difference is that the attacker also manages to elevate his privileges
```

File: rules/S2076/impact.adoc (hunk 8 above)
Comment (loris-s-sonarsource) [reply rules/S2076/impact.adoc:34]: you are right

**Pull request merged**